### PR TITLE
Flexy backend for +server

### PR DESCRIPTION
**Pull request by go-bears**
**Overview:**
update +server.py to take flexible backend for AI api calls.

**Changes:**
new files:
chat.ts
chat.test.ts
vitest.config.ts

modified files:
package.json (added dotenv package)
types.d.ts (added backend to GenerationRequest type)
ai.ts (updated with string builders)
ai.test.ts (updated with string builders)

Tested locally and seems to be working? 

can run with `npm run test`

### CODE REVIEW

File: src/lib/types.test.ts
Comment (nelson-lojo) [inline src/lib/types.test.ts:1]: All of these features are already provided by typescript checking ;; I'll remove this from the PR

**Comment by go-bears** [comment]
thanks for eyes @nelson-lojo ! After a good think and sleep, I'm wondering if generateObject needs to be revisited to allow for the schema free text generation?

**Comment by go-bears** [comment]
how to test without tests:
run app
`npm run dev`

in another terminal window run curl request (or use postman)

```bash
curl -X POST http://localhost:5173/api/object \
     -H "Content-Type: application/json" \
     -d '{
       "description": "Create a user object",
       "schema": "TestSchema",
       "typedef": "...",
       "backend": "cohere"
     }'
```

```bash
curl -X POST http://localhost:5173/api/object \
     -H "Content-Type: application/json" \
     -d '{
       "description": "Create a user object",
       "schema": "TestSchema",
       "typedef": "...",
       "backend": "openai"
     }'
```

**Comment by go-bears** [comment]
> thanks for eyes @nelson-lojo ! After a good think and sleep, I'm wondering if generateObject needs to be revisited to allow for the schema free text generation?

note: free form chat feature paused until after report is completed.

**Pull request merged**